pkg/graveler/graveler_test.go

```diff
@@ -352,6 +356,122 @@ func TestGraveler_Set(t *testing.T) {
 		})
 	}
 }
+
+func TestGravelerSet_Advanced(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	committedMgr := &testutil.CommittedFake{}
+	newSetVal := &graveler.ValueRecord{Key: []byte("key"), Value: &graveler.Value{Data: []byte("newValue"), Identity: []byte("newIdentity")}}
+	// RefManager mock base setup
+	refMgr := mock.NewMockRefManager(ctrl)
+	var getFlow = func() {
+		refMgr.EXPECT().ParseRef(gomock.Any()).Times(1).Return(graveler.RawRef{BaseRef: ""}, nil)
+		refMgr.EXPECT().ResolveRawRef(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(&graveler.ResolvedRef{
+			Type:                   0,
+			ResolvedBranchModifier: 0,
+			BranchRecord: graveler.BranchRecord{
+				BranchID: "",
+				Branch: &graveler.Branch{
+					CommitID:     "",
+					StagingToken: "",
+					SealedTokens: nil,
+				},
+			},
+		}, nil)
+		refMgr.EXPECT().GetCommit(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil, graveler.ErrCommitNotFound)
+	}
+
+	t.Run("update failure", func(t *testing.T) {
+		stagingMgr := &testutil.StagingFake{
+			UpdateErr: ErrGravelerUpdate,
+		}
+		refMgr.EXPECT().GetBranch(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(&graveler.Branch{}, nil)
+		getFlow()
+		store := newGraveler(t, committedMgr, stagingMgr, refMgr, nil, testutil.NewProtectedBranchesManagerFake())
+		err := store.Set(ctx, repository, "branch-1", newSetVal.Key, *newSetVal.Value, graveler.IfAbsent(true))
+		require.ErrorIs(t, err, ErrGravelerUpdate)
+		lastVal := stagingMgr.LastSetValueRecord
+		var expected *graveler.ValueRecord
+		require.Equal(t, expected, lastVal)
+	})
+
+	t.Run("branch deleted after update", func(t *testing.T) {
+		stagingMgr := &testutil.StagingFake{}
```

Comment:
Relevant to all tests - suggest to move it just before `newGraveler` as it is not relevant until that point.